### DISCUSSION TITLE
Where is the icon located for the tray icon?

### DISCUSSION THREAD

**Discussion by EaglePowers555** [BartoszCichecki/LenovoLegionToolkit#1510] [Q&A] [answered]
I'm a little OCD and like my tray icons to be black to match the system tray icons (windows 11). I have used resource hacker to change the icon library (all 5 icon sizes) in both the .exe and .dll. This does change the main program icon, but the tray icon persists as the red. Any help would be much appreciated!

**Comment by Ace-Radom** [comment, marked as answer]
Why not just compile the source code yourself with the dark icon you created? Since you already know how to use resource hacker it won't be so hard right? Basically you only need to replace the icon (`LenovoLegionToolkit.WPF/Assets/icon.ico`) with your own one, and run `make.bat` on a computer with develop environment installed and then, install your own version with the installer in the `build_installer` folder.

If you need help (for example how to install dev-env) I can provide it in this thread, but your question is actually something nobody cares about so I think no one knows it without a research.

**Reply by EaglePowers555** [reply]
Yeah, if you could walk me through how to compile the code that would be great

**Reply by Ace-Radom** [reply]
Basically you only need to download & install two sdks: [dotnet 8 sdk](https://dotnet.microsoft.com/en-us/download/dotnet/8.0) & [inno setup 6](https://jrsoftware.org/isdl.php). Remember install inno setup 6 to its default install directory (`C:\Program Files (x86)\Inno Setup 6`).

Then download the source code. You can download from [2.24.2 release](https://github.com/BartoszCichecki/LenovoLegionToolkit/releases/tag/2.24.2) or directly from repo site. There's no huge change.

After that go to source directory. You'll find a batch file named `make.bat`. Run it from command line or just double-click it, compilation will run automatically. Since you want a dark icon, replace the icon (`LenovoLegionToolkit.WPF/Assets/icon.ico`) before compiling.

And then just find the installer under `build_installer/` folder and install it. You may see the application name is something like `LenovoLegionToolkit [BETA]`, that's normal & won't effect anything. If you don't want this `BETA`, open command line under source directory and run the compile batch like `.\make.bat 2.24.2`. This will build a release with version num 2.24.2 and remove the beta sign.

**Reply by EaglePowers555** [reply]
Worked like a charm! Thank you so much
